internal/resources/grafana/resource_alerting_rule_group.go

```diff
@@ -604,6 +609,30 @@ func unpackRuleData(raw interface{}) ([]*models.AlertQuery, error) {
 	return result, nil
 }
 
+func checkFields(rule *models.ProvisionedAlertRule) error {
+	if rule.Record != nil {
+		incompatFieldMsgFmt := `"record" and "%s" cannot be set together`
+		if rule.For != nil {
+			return fmt.Errorf(incompatFieldMsgFmt, "for")
+		}
+		if rule.NoDataState != nil {
+			return fmt.Errorf(incompatFieldMsgFmt, "no_data_state")
+		}
+		if rule.ExecErrState != nil {
+			return fmt.Errorf(incompatFieldMsgFmt, "exec_err_state")
+		}
+		if rule.Condition != nil {
+			return fmt.Errorf(incompatFieldMsgFmt, "condition")
+		}
+	}
+
+	// condition is required if the rule is not a recording rule
+	if rule.Condition == nil {
+		return fmt.Errorf(`"condition" is required`)
+	}
+	return nil
+}
+
```

Comment:
I think you can achieve this via [schema attributes](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema#Schema) like `ConflictsWith` and `ExactlyOneOf` which will give you the `terraform validate` function and language server some teeth to help users

Reply:
Unfortunately sdkv2 limitations make this impossible on elements of a list type 😢